src/createServer.js

```diff
@@ -1,8 +1,51 @@
 'use strict';
 
+const fs = require('fs');
+const path = require('path');
+const http = require('http');
+
 function createServer() {
-  /* Write your code here */
-  // Return instance of http.Server class
+  const server = new http.Server();
+
+  server.on('request', async (req, res) => {
+    const url = new URL(req.url, `http://${req.headers.host}`);
+
+    if (req.method === 'POST' && url.pathname === '/add-expense') {
+      const chunks = [];
+
+      for await (const chunk of req) {
+        chunks.push(chunk);
+      }
+
+      const text = Buffer.concat(chunks).toString();
+      const data = JSON.parse(text);
```

Comment:
Parsing JSON without a try-catch block can lead to server crashes if the request body is not valid JSON. Consider wrapping `JSON.parse(text)` in a try-catch block to handle potential parsing errors gracefully.